.github/workflows/deployment.yml

```diff
@@ -28,7 +28,17 @@ on:
         default: true
 
 jobs:
+  validate-tag-name:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Validate tag name format
+        run: |
+          if [[ ! "${{ inputs.tag_name }}" =~ ^v[0-9]+\.[0-9]+\.[0-9]+$]]; then
```

Comment:
For those unfamiliar, see https://www.gnu.org/software/bash/manual/html_node/Conditional-Constructs.html

I have to keep confirming whether this is PCRE or POSIX _(its the latter hence why `\d` isn't doable)_.

Reply:
I think this is fine for now.  We might revisit this in the future when we decide we want to do release candidates or public previews of certain features; nothing definitive but an idea we've tossed around.